File: tiq-e/math/2/main.go
package main

import (
	"fmt"
	"time"
)

func countPrimes(n int) int {
	var cnt int
	data := make([]bool, n)
	for i := 2; i < n; i++ {
		if data[i] {
			continue
		}
		cnt++
		for j := i * i; j < n; j += i {
			data[j] = true
		}
	}
	return cnt
}

func main() {
	n := 15

	start := time.Now()
	result := countPrimes(n)
	end := time.Since(start)
	fmt.Println("Duration: ", end)
	fmt.Println("Result: ", result)
}

File: tiq-e/math/2/main.ts
import {performance} from "perf_hooks";

function countPrimes(n: number): number {
    let flagArray = [],
        result = 0;
    for (let i = 2; i < n; i++) {
        if (flagArray[i] === undefined) {
            flagArray[i] = 1;
            result++;
            let j = 2;
            while (i * j < n) {
                flagArray[i * j] = 0;
                j++;
            }
        }
    }
    return result;
}

const n = 15;

const start = performance.now();
const result = countPrimes(n);
const end = performance.now();
console.log("Duration:", end - start);
console.log("Result:", result);
